app/controllers/authenticated_controller.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# Provides authentication to derived controllers.
+# User is exposed via current_user method.
+class AuthenticatedController < ApplicationController
+  def current_user
+    @current_user ||= Authentication::UserById.new(
+      Authentication::IdFromJwt.new(
+        Http::BearerTokenFromHeaders.new(
+          request.headers
+        )
+      )
+    ).to_model
+  end
+
+  rescue_from Authentication::JwtCodec::InvalidTokenError, with: :not_authorized
+  rescue_from Http::BearerTokenFromHeaders::MissingToken, with: :not_authorized
+  rescue_from Http::BearerTokenFromHeaders::UnsupportedType,
+              with: :not_authorized
```

Comment:
@EgorBarakhov could you please help me with this? I try to handle errors gracefully, but server returns 500 error code anyway.

Reply:
I've created an issue to resolve it later - #30